### ISSUE TITLE
lookup + JSON template = unexpected type error

### ISSUE THREAD

**Issue by leedm777** [ansible/ansible#11885]
##### Summary:

When attempting to use `lookup('template', 'something.json.j2')`, I'm getting an `unexpected type error`. If I change the contents of `something.json.j2` to invalid JSON, the lookup works successfully.

I attempted adding Jinja2 comments to the file as a workaround, to no avail.
##### Issue Type:

Bug Report
##### Ansible Version:

Ansible 2 (commit e16b24c).
##### Ansible Configuration:

tasks/main.yml:

``` yaml
- debug: var='{{ lookup("template", "works.json.j2") }}'
  tags: debug

- debug: var='{{ lookup("template", "breaks.json.j2") }}'
  tags: debug
```

templates/works.json.j2:

``` json
x{ "works": "{{ ansible_date_time.date }}" }
```

templates/breaks.json.j2:

``` json
{ "breaks": "{{ ansible_date_time.date }}" }
```
##### Environment:

Python 2.7; Debian. Running within Docker.
##### Steps To Reproduce:
1. Create a role with the above config
2. Add it to a playbook
3. Run the play
##### Expected Results:

Run w/o errors.
##### Actual Results:

```
TASK [docker-registry : debug var={{ lookup("template", "works.json.j2") }}] ****
ok: [54.210.133.92] => {
    "changed": false, 
    "{{ lookup(\"template\", \"works.json.j2\") }}": "x{ \"foo\": \"2015-08-06\" }\n"
}

TASK [docker-registry : debug var={{ lookup("template", "breaks.json.j2") }}] ***
fatal: [54.210.133.92]: FAILED! => {"failed": true, "msg": "ERROR! an unexpected type error occurred. Error was sequence item 0: expected string, dict found"}
```

**Comment by MichaelDrogalis** [comment]
I've run into this one. Feature is unusable as in.

**Comment by vzaharee** [comment]
The issue actually occurs if the template is valid **Python** data, which also happens to be JSON that doesn't use `true`, `false`, or `null`.

**Comment by markosamuli** [comment]
I've run into this when using `iam_policy` with a JSON template. What seems to work for me is to just add a line break into the beginning of the JSON template and it gets cleaned up in the process.

Using ansible-playbook 2.0.0 (devel 9bb95b5235)

**Comment by jensschroer** [comment]
Just stumbled across the same error.
ansible 2.0.0 (devel 404ded2b03) last updated 2015/10/22 16:34:33 (GMT +200)
  lib/ansible/modules/core: (detached HEAD bc3b1abd68) last updated 2015/10/22 16:34:45 (GMT +200)
  lib/ansible/modules/extras: (detached HEAD 405c3cb2c7) last updated 2015/10/22 16:34:45 (GMT +200)
  config file = /home/schroer/.ansible.cfg
  configured module search path = None

**Comment by jensschroer** [comment]
I just added a pull request for this issue. If you apply the patch and modify the line
{{ lookup("template", "breaks.json.j2") }}'
to
{{ lookup("template", "breaks.json.j2",convert_data=False) }}'
it should work.
@leedm777 any chance you could confirm?

**Comment by fperks** [comment]
@jensschroer 

I can confirm it works for me. Thanks for the fix.

**Comment by leedm777** [comment]
Sorry for the delay.

Confirmed fix with 2.0.0.2. Thanks!
